pkg/apis/openebs.io/v1alpha1/common_volume_properties.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// CapacitySpec wraps the Capacity property
+//
+// NOTE: It is intended to be used as embedded struct
+type CapacitySpec struct {
```

Comment:
request @kmova @vishnuitta @avishnu to have a look at this schema design.
IMO we need not have so much finer granularity.
We might want to have ISCSISpec, NFSSpec, or BlockVolumeSpec, FileVolumeSpec kind of structures.
All the above are common structures that should be embedded into CASVolume struct. A CASVolume struct can be any of the above.

Above was my thought process.